### PR TITLE
Millestone 4: Connecting To API with Redux

### PR DESCRIPTION
**Pull request by skabeo**
In this pull request, the `createAsyncThunk` package was introduced.

- The `createAsyncThunk` made it posisble to create a separate action creator function.
- The function takes in an action type as an argument and a call back function.
- In the callback function an async function can be created to make request to an API.
- The `createasyncThunk` was exported to the `Bookrender.js` to get data from the API whenever the component is mounted.
- Finally `extraReducers` were created with three diffrent `addCases` to change our state based on the result of the `createAsymcThunk` call. It would either be `pending, fulfilled or rejected`. Based on this our entire state can be set to change.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,41 +1,65 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable camelcase */
```
Comment (Deepakdanger) [inline src/redux/books/booksSlice.js]: - [ ] kindly remove this line & solve the linter manually.

File: src/components/Bookrender.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
+/* eslint-disable camelcase */
```
Comment (Deepakdanger) [inline src/components/Bookrender.js]: - [ ] kindly remove this line & solve the linter manually.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @skabeo , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @skabeo 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**